### PR TITLE
Fix Chrome app and SSL login

### LINKED ISSUE #657: Login fails in Chrome app
In `git master`. When pressing `Login` in the Chrome app, then I get passed to Github for auth, but upon completing I'm redirected to what seems to be `localhost:9999/` instead of whatever the Chrome app url thing should be.

### PR DESCRIPTION
**Pull request by bergie**
Fixes #657

### CODE REVIEW

File: components/LoginOAuth.coffee [hunk 1]
```diff
@@ -2,14 +2,19 @@ noflo = require 'noflo'
 url = require 'url'
 qs = require 'querystring'
 
-isRedirectValid = (redirect) ->
+isRedirectValid = (redirect, chrome) ->
   parsedRedirect = url.parse redirect
-  parsedAppRedirect = url.parse '$NOFLO_OAUTH_CLIENT_REDIRECT'
+  if chrome
+    parsedAppRedirect = url.parse '$NOFLO_OAUTH_CHROME_CLIENT_REDIRECT'
```
Comment (jonnor) [inline components/LoginOAuth.coffee]: Can't wait to get rid of this shitty string replacement stuff...

File: Gruntfile.coffee [hunk 2]
```diff
@@ -147,6 +147,30 @@ module.exports = ->
             pattern: /\$NOFLO_OAUTH_CLIENT_SECRET/ig
             replacement: process.env.NOFLO_OAUTH_CLIENT_SECRET or ''
           ,
+            pattern: /\$NOFLO_OAUTH_SSL_CLIENT_ID/ig
+            replacement: process.env.NOFLO_OAUTH_SSL_CLIENT_ID or ''
+          ,
+            pattern: /\$NOFLO_OAUTH_SSL_CLIENT_REDIRECT/ig
```
Comment (jonnor) [inline Gruntfile.coffee]: Why do we need separate settings for SSL? What does it mean if these values are empty (the default)?

File: manifest.dist.json [hunk 3]
```diff
@@ -9,8 +9,9 @@
   },
   "permissions": [
     "$NOFLO_OAUTH_GATE/",
-    "$NOFLO_REGISTRY_SERVICE",
+    "$NOFLO_REGISTRY_SERVICE/",
```
Comment (jonnor) [inline manifest.dist.json]: This seems unrelated to login fixes, why the change?

Comment (bergie) [inline manifest.dist.json]: Apparently Chrome app CSP URLs require a trailing slash

File: Gruntfile.coffee (hunk 2 above)
Comment (bergie) [inline Gruntfile.coffee]: Because GitHub OAuth doesn't accept http and https callbacks for same app, so you need two apps

**Pull request merged**